src/jetstream/types.rs

```diff
@@ -710,6 +738,37 @@ pub(crate) struct StreamNamesResponse {
     pub streams: Vec<String>,
 }
 
+/// Options to configure Pull Subsscription
+#[derive(Debug, Default, Clone)]
+pub struct PullSubscibeOptions {
+    pub(crate) stream_name: Option<String>,
+    pub(crate) bind_only: bool,
+    pub(crate) consumer_config: Option<ConsumerConfig>,
+}
+
+impl PullSubscibeOptions {
+    /// creates new options
+    pub fn new() -> PullSubscibeOptions {
+        Default::default()
+    }
+
+    /// Binds subscription explicitly to a stream.
+    /// If not specified, stream will be looked up based on the subject provided.
+    pub fn bind_stream(mut self, stream_name: String) -> Self {
+        self.stream_name = Some(stream_name);
+        self.bind_only = true;
+        self
+    }
+
+    /// when creating Pull Subscription to not existing consumer
+    /// Consumer Configuration can be specified
+    /// will apply only if `bind_stream` is not called and consumer doesn't exist.
+    pub fn consumer_config(mut self, consumer_config: ConsumerConfig) -> Self {
```

Comment:
Diverges quite a bit from regular SubscribeOptions, in that it gives you per field accessors.
Is this enough? Could we just reuse the same options and enforce a different validation?

Reply:
I wanted to avoid ability to pass wrong config, as it confused me few times and enforce compile-time correctness. That said, I like neither outcome. Will think about it while adding new features from 2.7.0 server release.

Reply:
Unless @derekcollison prefers to just have the general subscription options and handle missconfig while calling `pull_subscribe`.

Reply:
We are looking to simplify the clients around JS consumers, so be good to touch base here.